### PR TITLE
Full Site Editing: Rename the Post Content and Template blocks

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request

* Rename `a8c/content-slot` into `a8c/post-content`
* Rename `a8c/template-part` into `a8c/template`

Note: this PR doesn't change the functionality, nor the copy.
Let's give us more time to think how we want to present these blocks to the users, and involve editorial when necessary.

See paAmJe-fT-p2

#### Testing instructions

* [Build the Full Site Editing plugin](https://github.com/Automattic/wp-calypso/blob/ec422edf146a4d0e2c46a9515a65058d8f1d4950/apps/full-site-editing/README.md#local-development).
* Smoke test the Content Slot and Template Part blocks and make sure they work as expected:
  - Content Slot should not render anything on the front end (just a temporary placeholder message), and shouldn't persist the selected content preview.
  - Template Part should render the selected content and persist it in the editor.
* Switch to code editor and make sure they appear as `a8c/post-content` and `a8c/template`.

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/blocks/template/index.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { __ } from '@wordpress/i18n';
 import edit from './edit';
 import './style.scss';
 
-registerBlockType( 'a8c/template-part', {
+registerBlockType( 'core/template', {
```
Comment (vindl) [inline apps/full-site-editing/full-site-editing-plugin/blocks/template/index.js]: This name is already taken in core :/ https://github.com/WordPress/gutenberg/blob/0fce12abc365ba85288bdb71b9115d598776fcb4/packages/block-library/src/template/block.json#L2, which results in the following console error: `Block "core/template" is already registered.`

**Comment by vindl** [comment]
Spotted one more console warning while testing that's coming from `PostAutocomplete` component: ```Warning: Invalid DOM property `autocomplete`. Did you mean `autoComplete`?```

There are a couple more console warnings that pop up when using the Post Content block.

File: apps/full-site-editing/full-site-editing-plugin/blocks/template/index.js (hunk 1 above)
Comment (Copons) [reply apps/full-site-editing/full-site-editing-plugin/blocks/template/index.js]: :oldmanyellsatcloud: 😄

**Comment by Copons** [comment]
> Spotted one more console warning while testing that's coming from `PostAutocomplete` component: `` Warning: Invalid DOM property `autocomplete`. Did you mean `autoComplete`? ``
> 
> There are a couple more console warnings that pop up when using the Post Content block.

Good catch, I forgot to `define('SCRIPT_DEBUG', true);` and so my dev site was running with prod React which suppresses that warning!

File: apps/full-site-editing/full-site-editing-plugin/blocks/template/index.js (hunk 1 above)
Comment (mattwiebe) [reply apps/full-site-editing/full-site-editing-plugin/blocks/template/index.js]: There is a `unregisterBlockType` function so we could run that just before re-registering our block.

Comment (Copons) [reply apps/full-site-editing/full-site-editing-plugin/blocks/template/index.js]: If we aim to eventually bring this into Core, I don't think unregistering Core blocks first would be an ideal approach. 😄

**Comment by mattwiebe** [comment]
I'm just gonna kick this back to In Progress since we're still getting this error `Block "core/template" is already registered.`

**Comment by Copons** [comment]
It was decided to keep namespacing these with `a8c`, so the final names expected from this PR will be:

- `a8c/post-content`
- `a8c/template`

**Pull request merged**